### DISCUSSION TITLE
Provide a 'build status' or 'build health' for each software package

### DISCUSSION THREAD

**Discussion by MikeFultonDev** [zopencommunity/meta#35] [Ideas]
At any given point, various software packages being ported to z/OS are at various levels of quality from 'works well, most (all) tests pass' to 'does not even build'.
@IgorTodorovskiIBM put together a slide with a 'point in time' view of this, but I'm thinking it would be nice to have a way for the software package repositories to describe this about themselves, and then we could have a tool that could provide a general report and such so people know what doesn't work without having to dig through issues?

@AnthonyGiorgio your comment that 'vim does not seem to work' (which is true) is what got me thinking about this...

**Comment by IgorTodorovskiIBM** [comment]
One option is to add the requirement of a zopen "status" file in each port repo.  zopen could look for this file and generate it as a package artifact. Another option is to base it on the portchk.sh script.

We could then propagate the "status" to this page https://zosopentools.github.io/meta/#/Latest

**Reply by MikeFultonDev** [reply]
I like the idea of a 'status' file - can we change the semantics of portchk.sh to be responsible for generating the file? And perhaps we should switch over the Jenkins CI/CD to now start running tests?

**Reply by IgorTodorovskiIBM** [reply]
I have a draft PR here: https://github.com/ZOSOpenTools/utils/pull/52 . I decided to use the return code from the check script to drive the generation of the build.status file.

**Comment by MikeFultonDev** [comment]
done.
